submissions/chernetskyi8704/Tiny JS World/index.js

```diff
@@ -0,0 +1,199 @@
+class Inhabitant {
+  constructor(Species, Name, Gender, Saying) {
+    this.Species = Species;
+    this.Name = Name;
+    this.Gender = Gender;
+    this.Saying = Saying;
+  }
+
+  addBoldAndRedSemicolon(string) {
+    return string.bold().fontcolor("red");
+  }
+
+  addRedBoldCollon(string) {
+    return string.bold().fontcolor("red");
+  }
+
+  makeStringRedAndBold(string) {
+    return string.bold().fontcolor("Red");
+  }
+
+  makeStringBold(string) {
+    return string.bold();
+  }
+
+  saying() {
+    return ["Saying"].map(
+      (propertyOfTheInhabitant) =>
+        ` ${
+          this.makeStringRedAndBold(propertyOfTheInhabitant) +
+          this.addRedBoldCollon(":")
+        } ${this.makeStringBold(this.Saying)}` +
+        this.addBoldAndRedSemicolon(";")
+    );
+  }
+
+  displayData() {
+    return (
+      ["Species", "Name", "Gender"]
+        .map(
+          (propertyOfTheInhabitant) =>
+            `${
+              this.makeStringRedAndBold(propertyOfTheInhabitant) +
+              this.addRedBoldCollon(":")
+            } ${this[propertyOfTheInhabitant].bold()}`
+        )
+        .join(this.addBoldAndRedSemicolon("; ")) +
+      this.addBoldAndRedSemicolon(";")
+    );
+  }
+}
+class Human extends Inhabitant {
+  constructor(Species, Name, Gender, Saying) {
+    super(Species, Name, Gender, Saying);
+    this.Hands = "2";
+    this.Legs = "2";
+  }
+
+  displayData() {
+    return (
+      `${super.displayData()} ` +
+      ["Hands", "Legs"]
+        .map(
+          (propertyOfTheInhabitant) =>
+            `${
+              this.makeStringRedAndBold(propertyOfTheInhabitant) +
+              this.addRedBoldCollon(":")
+            } ${this[propertyOfTheInhabitant].bold()}`
+        )
+        .join(this.addBoldAndRedSemicolon(";")) +
+      this.addBoldAndRedSemicolon("; ") +
+      this.saying()
+    );
+  }
+}
+
+class Animal extends Inhabitant {}
+
+class Man extends Human {
+  constructor(Name, Gender, Saying) {
+    super("Man🤵", Name, Gender, Saying);
+  }
+}
+
+class Woman extends Human {
+  constructor(Name, Gender, Saying) {
+    super("Woman👩", Name, Gender, Saying);
+  }
+}
+
+class Dog extends Animal {
+  constructor(Name, Gender, Saying) {
+    super("Dog🐶", Name, Gender, Saying);
+    this.Paws = "4";
+    this.Tail = "1";
+  }
+
+  displayData() {
+    return (
+      `${super.displayData()} ` +
+      ["Paws", "Tail"]
+        .map(
+          (propertyOfTheInhabitant) =>
+            `${
+              this.makeStringRedAndBold(propertyOfTheInhabitant) +
+              this.addRedBoldCollon(":")
+            } ${this[propertyOfTheInhabitant].bold()}`
+        )
+        .join(this.addBoldAndRedSemicolon("; ")) +
+      this.addBoldAndRedSemicolon(";") +
+      this.saying()
+    );
+  }
+}
+
+class Cat extends Animal {
+  constructor(Name, Gender, Saying) {
+    super("Cat🐱", Name, Gender, Saying);
+    this.Paws = "4";
+    this.Tail = "1";
```

Comment:
Not critical here, as we do not really do math operations with paws and tails.
Just wondering, why strings?

Reply:
Strings because of using string methods. 
Should i fix it?

Reply:
Not necessarily. For example, telephone numbers do not actually make sense as numbers in mathematial sense, as we do not math operations over them. From the other hand we may want to count total number of legs in a herd.

Probably a missing part in your puzzle was that numbers coerce to strings automagically in string context.
Check what `4 + ""` results in and why `[1, 31, 3, 5, 11].sort()` doesn't work as expected. This is an important part of understanding JS internals and peculiarities.

You may skip fixing this. Make sure to take some learnings.

Reply:
Type conversion it's really a great thing in JS but I'm not thought about this in the border of this task. In the future will ask my self more often 'What about if...'. Thank you so much for the review.